### PR TITLE
Feature: Files Search Schema and stub implementation [ch19245]

### PR DESCRIPTION
**Pull request by perfectmak**
**CHANGELOG**
- Create schema for searching files through a `SearchFiles` rpc method.
- Currently, the default search implementation just returns list of file entries in the root path of the current users default bucket. This implementation would be fixed in the upcoming PR's.
- Refactors the logic to convert a domain `FileInfo` into `ListDirectoryEntry`. To reduce duplicate code of the same logic

### CODE REVIEW

File: grpc/proto/space.proto [hunk 1]
```diff
@@ -308,6 +308,23 @@ service SpaceApi {
       body: "*"
     };
   }
+
+  // Search for files across all users bucket
+  rpc SearchFiles(SearchFilesRequest) returns (SearchFilesResponse) {
+    option (google.api.http) = {
+      post: "/v1/search/files"
```
Comment (jsonsivar) [inline grpc/proto/space.proto:315]: Think this should be `get`

**Review by gpuente — APPROVED** [review]
LGTM

**Review by jsonsivar — COMMENTED** [review]
LGTM just needs to be a `GET`.

File: grpc/proto/space.proto (hunk 1 above)
Comment (perfectmak) [reply grpc/proto/space.proto]: True.... I've fixed it.

**Review by maurycy — APPROVED** [review]
LGTM. 🚀

**Pull request merged**